src/disk_io_thread.cpp

```diff
@@ -1633,7 +1625,6 @@ namespace libtorrent
 			j->error.operation = storage_error::read;
 			return 0;
 		}
-		j->flags |= disk_io_job::use_disk_cache;
```

Comment:
removing that block would make use_read_cache = false meaningless though, wouldn't it? I was imagining that the semantics would be, read from the cache if there happens to be something there, otherwise, issue an uncached read.

Do think it would work to remove that assert instead? or turn it into a do_uncached_read()?

In theory there's a risk that a write job can slip in while this read job is in the queue, and then the read job will get outdated data. I'm pretty sure that can't happen though, because of the way pieces are written. They're always written before read, except in the case of a failing hash check, but then the queue is flushed for that storage to start over.